### PR TITLE
[SPARK-38186][SQL] Improve the README of Spark docs

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Improve https://github.com/apache/spark/blob/master/docs/README.md, mark some of the setup steps as optional.
Also, recommend developers to use `SKIP_API=1` if no API docs are needed.

### Why are the changes needed?

 Developers usually need to generate HTML Docs without API Docs. Marking the setup for SQL/Python/R doc as optional can help them avoid unnecessary efforts.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manual test

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @xinrong-databricks who just managed to set up doc generation as well.

**Review by xinrong-meng — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -48,23 +48,9 @@ $ bundle install
 
 Note: If you are on a system with both Ruby 1.9 and Ruby 2.0 you may need to replace gem with gem2.0.
```
[the remark this answers is not in the document]
Comment (gengliangwang) [reply docs/README.md:49]: I am not very sure about this. Are you using M1 Mac or Intel?

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: docs/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (xinrong-databricks) [reply docs/README.md:49]: Intel processor, and Big Sur macOS. The current change looks good enough :)

**Pull request closed without merging**